pkg/provider/azure_vmssflex_cache.go

```diff
@@ -132,26 +176,34 @@ func (fs *FlexScaleSet) getNodeVmssFlexID(nodeName string) (string, error) {
 		return fmt.Sprintf("%v", cachedVmssFlexID), nil
 	}
 
-	cached, err := fs.vmssFlexCache.Get(consts.VmssFlexKey, azcache.CacheReadTypeDefault)
-	if err != nil {
-		return "", err
-	}
-	vmssFlexes := cached.(*sync.Map)
-
-	vmssFlexes.Range(func(key, value interface{}) bool {
-		vmssFlexID := key.(string)
-		_, err := fs.vmssFlexVMCache.Get(vmssFlexID, azcache.CacheReadTypeForceRefresh)
+	getter := func(nodeName string, crt azcache.AzureCacheReadType) (string, error) {
+		cached, err := fs.vmssFlexCache.Get(consts.VmssFlexKey, crt)
 		if err != nil {
-			klog.V(12).Infof("failed to refresh vmss flex VM cache for vmssFlexID %s", vmssFlexID)
+			return "", err
 		}
-		return true
-	})
+		vmssFlexes := cached.(*sync.Map)
 
-	cachedVmssFlexID, isCached = fs.vmssFlexVMNameToVmssID.Load(nodeName)
-	if isCached {
-		return fmt.Sprintf("%v", cachedVmssFlexID), nil
+		vmssFlexes.Range(func(key, value interface{}) bool {
+			vmssFlexID := key.(string)
+			_, err := fs.vmssFlexVMCache.Get(vmssFlexID, azcache.CacheReadTypeForceRefresh)
+			if err != nil {
+				klog.V(12).Infof("failed to refresh vmss flex VM cache for vmssFlexID %s", vmssFlexID)
+			}
+			return true
+		})
+
+		cachedVmssFlexID, isCached = fs.vmssFlexVMNameToVmssID.Load(nodeName)
+		if isCached {
+			return fmt.Sprintf("%v", cachedVmssFlexID), nil
+		}
+		return "", cloudprovider.InstanceNotFound
 	}
-	return "", cloudprovider.InstanceNotFound
+
+	vmssFlexID, err := getter(nodeName, azcache.CacheReadTypeDefault)
+	if err == cloudprovider.InstanceNotFound {
+		return getter(nodeName, azcache.CacheReadTypeForceRefresh)
```

Comment:
add a `klog.V(2).Infof` logging before cache refresh, that would benefit in troubleshooting

Reply:
Good point. Just added the logs